src/constructs/autoscaling/asg.test.ts

```diff
@@ -40,7 +40,21 @@ describe("The GuAutoScalingGroup", () => {
     });
   });
 
-  test("adds the instanceType parameter", () => {
+  test("does not add the AMI parameter if an imageId prop provided", () => {
```

Comment:
I believe that Riff-Raff [currently relies on AMI being configured as a parameter](https://riffraff.gutools.co.uk/docs/magenta-lib/types#amicloudformationparameter) in order to automatically update the underlying AMI during a deployment. That's a widely used feature in the department (and ideally we want every application to do this due to the security benefits it brings), so I think we might need to discourage passing it as a prop for now.

Reply:
That's a good point. The reason this was added back as an optional prop was to allow for stacks that have multiple ASGs. Currently, that would fail as we'd try to create the same parameter more than once. This would allow a stack to define multiple parameters and pass them into each ASG as required. If we want to go as far as requiring AMI values to be a parameter, we could change the type of `imageId` to `GuAMIParameter` to allow for this overriding but prevent the value being hardcoded in the stack.

Reply:
> The reason this was added back as an optional prop was to allow for stacks that have multiple ASGs. 

Hmm, I hadn't thought about that other use-case! Maybe we should still allow this then and use patterns/docs to push 'typical' single-ASG stacks towards using an AMI parameter.

Reply:
I've added a suggested below which might be the best of both worlds. Agreed that the standard EC2 pattern should then just create the parameter for you!